### DISCUSSION TITLE
How to configure Ruffle into Chromium?

### DISCUSSION THREAD

**Discussion by Varaosa** [ruffle-rs/ruffle#3692] [Q&A]
Hi,

I just want to ask how to configure Ruffle support into Chromium if, you don't have Chromium installed on your pc?

I have one application that use Chromium system & allow to use DLL with CEF command line but, because i don't have Chromium self installed on my pc because, i don't use Chromium, i have FireFox & there Ruffle work but, not in Chromium, i can't adjustment Chromium it self because, i don't have Chromium so, i need someshort DLL -file or pluging that allow to use Ruffle in Chromium, when play flash in Chromium worked app.

Im already ask this question into those application support about Flash but, they say just it's all about & issue in Ruffle so, im here for now to ask this then, how i can solute this issue to configure Ruffle to work in Chromium if, i don't even have Chromium installed on my pc. x3

**Comment by Poopooracoocoo** [comment]
Oh so you're talking about [CEF](https://bitbucket.org/chromiumembedded/cef/src/master/)?

**Reply by Varaosa** [reply]
I dunno or yet. Application support just say then if you want to get Ruffle work in those application you need someshort command or use CEF script to use Ruffle if you don't have Chromium installed your pc. 

For additional i need Ruffle work in Chromium when flash was in "html" -file. Those html file loaded into Chromium browser or in this case in those application that use someting Chromium & ruffle & flash need to loaded there.

I was try that but, i got error:

Someting went wrong :c

It appears you running Ruffle as the "file:" protocol.
This doesn't work as browsers block many fatures from working for security reasons.
Instaead, we invite you setup a local server or either use the web demo or desktop application.

-This not happend in FireFox, that i use mainly. I got IE too as usual, there ruffle not working or flash but, i don't currently need that here, i just need ruffle html+flash working in Chromium, that i don't have.

**Comment by Varaosa** [comment]
I still don't know how to fix this & i still got issue with Ruffle in Chromium system.

**Reply by DaVince** [reply]
It sounds to me like you are not looking to integrate Ruffle into a browser that you don't use, but that you just want to provide support on a HTML page for Ruffle itself. Ruffle has a JavaScript + WebAssembly variant which you can add straight into your HTML file and all browsers that support WebAssembly will just pick up on it.

* First, [download the latest self-hosted version of Ruffle](https://ruffle.rs/#downloads).
* Extract the package wherever your HTML and SWF files are.
* Then, [follow the basic instructions for website integration](https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#web).

You will have to serve your HTML file, along with the SWF file and Ruffle, from a web server instead of opening the HTML file directly with a browser. There are plenty of options for that. Two simple debug HTTP servers would be the Node-based [http-server](https://www.npmjs.com/package/http-server) or Python's [http.server](https://docs.python.org/3/library/http.server.html). Serve up the HTML file along with the other files and point your browser (embedded? headless? something else?) to it.

**Reply by Varaosa** [reply]
> 
> 
> It sounds to me like you are not looking to integrate Ruffle into a browser that you don't use, but that you just want to provide support on a HTML page for Ruffle itself. Ruffle has a JavaScript + WebAssembly variant which you can add straight into your HTML file and all browsers that support WebAssembly will just pick up on it.
> 
>     * First, [download the latest self-hosted version of Ruffle](https://ruffle.rs/#downloads).
> 
>     * Extract the package wherever your HTML and SWF files are.
> 
>     * Then, [follow the basic instructions for website integration](https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#web).
> 
> 
> You will have to serve your HTML file, along with the SWF file and Ruffle, from a web server instead of opening the HTML file directly with a browser. There are plenty of options for that. Two simple debug HTTP servers would be the Node-based [http-server](https://www.npmjs.com/package/http-server) or Python's [http.server](https://docs.python.org/3/library/http.server.html). Serve up the HTML file along with the other files and point your browser (embedded? headless? something else?) to it.

-No, i exacly want to integrate ruffle into a browser that i don't use. I have no issue to use ruffle into hosted version or local in wanted web-browser that i use but, i have issue to use ruffle with Chromium that i don't have.

The real problem here is that, I have an application (Wallepaper Engine) that runs or uses a Chromium based system that is in the program itself. In order to work with flash ruffle and this application, your html file must be Chromium compatible and I can’t or won’t do this because I don’t own a Chromium browser. But, I can get flash works in html in other browsers but, I don't get this to work in Chromium. 

The problem here is also that, I can't adjust or do anything myself in the (Wallepaper Engine) -application, that I use, this has to be done using a ruffle and/or with html file, that use ruffle engine.

**Comment by Varaosa** [comment]
I still have issue with this & my projects.

I don't have currently any issue with Ruffle to use in html or play flash in locally in html using Ruffle in FireFox but, issue is in other web browsers or softwares, that or where i want to use this project.

My current project need to go Wallepaper Engine that based Chromium. Are there any sample html project or script that use Ruffle, that are made or based Chromium or & can work in Microsoft Edge, that support or work for now i guess with Chromium?
